homeassistant/config.py

```diff
@@ -1025,21 +1046,167 @@ async def merge_packages_config(
+        ),
+        "config_schema_unknown_err": (
+            f"Unknown error calling {domain} CONFIG_SCHEMA",
+            True,
+        ),
+        "platform_validator_unknown_err": (
+            f"Unknown error validating {platform_name} platform config with {domain} "
+            "component platform schema",
+            True,
+        ),
+        "platform_component_load_err": (f"Platform error: {domain} - {ex}", False),
+        "platform_component_load_exc": (f"Platform error: {domain} - {ex}", True),
+        "platform_schema_validator_err": (
+            f"Unknown error validating config for {platform_name} platform "
+            f"for {domain} component with PLATFORM_SCHEMA",
+            True,
+        ),
+    }
+    return log_message_mapping.get(translation_key, (None, False))
+
+
+async def async_process_component_and_handle_errors(
+    hass: HomeAssistant,
+    config: ConfigType,
+    integration: Integration,
+    raise_on_failure: bool = False,
 ) -> ConfigType | None:
-    """Check component configuration and return processed configuration.
+    """Process and component configuration and handle errors.
+
+    In case of errors:
+    - Print the error messages to the log.
+    - Raise a ConfigValidationError if raise_on_failure is set.
+
+    Returns the integration config or `None`.
+    """
+    integration_config_info = await async_process_component_config(
+        hass, config, integration
+    )
+    return async_handle_component_errors(
+        hass, integration_config_info, integration, raise_on_failure
+    )
 
-    Returns None on error.
+
+@callback
+def async_handle_component_errors(
+    hass: HomeAssistant,
+    integration_config_info: IntegrationConfigInfo,
+    integration: Integration,
+    raise_on_failure: bool = False,
+) -> ConfigType | None:
+    """Handle component configuration errors from async_process_component_config.
+
+    In case of errors:
+    - Print the error messages to the log.
+    - Raise a ConfigValidationError if raise_on_failure is set.
+
+    Returns the integration config or `None`.
+    """
+
+    processed_config = integration_config_info.config
+    config_exception_info = integration_config_info.exception_info_list
+    if not config_exception_info:
+        return processed_config
+
+    platform_exception: ConfigExceptionInfo
+    domain = integration.domain
+    placeholders: dict[str, str] = {
+        "domain": domain,
+    }
+    for platform_exception in config_exception_info:
+        exception = platform_exception.exception
+        placeholders["error"] = str(exception)
+        platform_name = placeholders["p_name"] = platform_exception.platform_name
+        platform_config = platform_exception.config
+        log_message, show_stack_trace = _get_log_message_and_stack_print_pref(
+            platform_exception.translation_key, exception, domain, platform_name
+        )
+        link = platform_exception.integration_link or integration.documentation
```

Comment:
Why not always include the link to the documentation in `ConfigExceptionInfo`?
```suggestion
        link = platform_exception.integration_link
```